src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs

```diff
@@ -182,13 +183,24 @@ public BitmapSource FinishImageProcessing(BitmapImage iconBitmapImage, Uri iconU
         }
 
         private static void AddToCache(Uri iconUrl, BitmapSource iconBitmapImage)
+        {
+            string cacheKey = GenerateKeyFromIconUri(iconUrl);
+            AddToCache(cacheKey, iconBitmapImage);
+        }
+
+        private static void AddToCache(string cacheKey, BitmapSource iconBitmapImage)
         {
             var policy = new CacheItemPolicy
             {
                 SlidingExpiration = TimeSpan.FromMinutes(10),
                 RemovedCallback = CacheEntryRemoved
             };
-            BitmapImageCache.Set(iconUrl.ToString(), iconBitmapImage, policy);
+            BitmapImageCache.Set(cacheKey, iconBitmapImage, policy);
```

Comment:
I'm only just glancing at the PR now, but won't this overwrite images if there is more than 1 non-image icon?
Both will get a key of `string.Empty`, so they won't be unique. Should we generate a GUID or similar?

Reply:
The hash key is generated internally in BitmapImageCache.

Suppose you have 2 non-image icons with different non-null URIs. Both icons will fail loading and add the default icon image with their respective Uri